src/gui/plugins/apply_force_torque/ApplyForceTorque.cc

```diff
@@ -209,7 +211,19 @@ ApplyForceTorque::ApplyForceTorque()
 }
 
 /////////////////////////////////////////////////
-ApplyForceTorque::~ApplyForceTorque() = default;
+ApplyForceTorque::~ApplyForceTorque()
+{
+  if (!this->dataPtr->scene)
+    return;
+  this->dataPtr->forceVisual->SetVisible(false);
+  this->dataPtr->forceVisual->Destroy();
+
+  this->dataPtr->torqueVisual->SetVisible(false);
+  this->dataPtr->torqueVisual->Destroy();
+
+  this->dataPtr->gizmoVisual->SetVisible(false);
+  this->dataPtr->gizmoVisual->Destroy();
```

Comment:
ah yes this works but prefer to use `Scene::DestroyNode` as it ensures the the visuals are also removed from the scene's internal list of nodes: 

```suggestion
  this->dataPtr->scene->DestroyNode(this->dataPtr->forceVisual, true);
  this->dataPtr->scene->DestroyNode(this->dataPtr->torqueVisual, true);
  this->dataPtr->scene->DestroyNode(this->dataPtr->gizmoVisual, true);

```

Reply:
Alright, changed it in db942e4